### ISSUE TITLE
Taming Horse

### ISSUE THREAD

**Issue by lawrencekellie** [scalda/Foolcraft_3#359]
## Issue Description
When trying to tame a horse by empty hand right mouse clicking on the horse to get on it and then get bucked off, I never get on the horse. I am teleported to a nearby location.

## Reproduce Steps
find horse
Make sure you have an empty hand.
Right mouse click on horse.

**Comment by Revvilo** [comment]
Are you quite literally teleported to another location, or is your camera just turned? Because I know of an issue, which I can't quite remember the cause of, which makes your game think that it's going to get on the horse, but then the server decides otherwise, and thus your camera is turned and nothing happens.
My first thought is to just make sure that you have both hands empty. Having something in your left hand often breaks things that need an empty hand to use.

**Comment by lawrencekellie** [comment]
Thank you for your response.

I'm really not sure what you mean by camera turned on. so let me explain
this in a little different way.

When you go to tame a horse, and you click on the horse and end up on its
back, if the horse isn't tamed, you get bucked off. My phrase teleported
off I could apply to the above sentence. In other words, I am at one
location when I click on the horse and immediately in another, right beside
the horse, as if I *had* been bucked off.

Both hands are empty. I almost never have something in my off hand.

Just for the fun of it, I had the thought that maybe if I clicked on the
horse enough times, it would eventually become tamed. So I did that,
clicking 20 times. The horse was not tamed at the end. But I did notice
something else and that is that I always “landed” in the same spot.

Hope that helps.

**Comment by Revvilo** [comment]
Ahh, I see. That's actually very weird. What I meant was that in certain situations your camera can be rotated (instantaneously) without actually getting on the horse, but it doesn't sound like that's what you're experiencing.
Have you tried relogging since you started having the problem? Because it really sounds like what happens if you're holding the crouch button when you get on a mount. Maybe the game has glitched and thinks you're still crouching.

**Comment by lawrencekellie** [comment]
Thank you.

I'm done playing for today. 😃 I'll try tomorrow. I also have another idea
that I want to try.

This situation is not critical for me, as I very seldom use horses, but I
do like having one tamed just for "recreational" riding.  😘

Lawrence.

**Comment by lawrencekellie** [comment]
One thing that I forgot to mention.  I've played many, many modpacks.  This
one is one of the best, if not the best.  Absolutely rates in the "best"
category!


I've decided to do a reinstall of the modpack to see if I can duplicate the
problem in a new install.

Thank you.

Lawrence

**Comment by lawrencekellie** [comment]
Test and Results

I did a fresh install of FoolCraft 3, started it up (play), and created a
new world.  The name of the world was OneBanana.  I chose "Cheats," which I
always do, and then world = Biomes of Plenty.  In the "Seed" box, I put
OneBanana...yes the same as the world name.  I then created the world.

The landing for the spawn point was the same as before.  However, my "base"
was at another location:  -258,65,-5443.  I teleported there (I have
cheats, remember?)  I then looked for the horse that I had tried before.  I
found it, but it was not mature yet.  It is black with white spots, with a
black colored mother.  It is near the coordinates -320 57 -5356.  I clipped
into creative and got 64 golden apples, the first version, not the more
powerful one.  I took 64 because I didn't know if the amount would diminish
when I fed them to the colt.  I continued until it was a mature horse.  I
slipped back into survival, moved to a point on the hotbar so I was not
holding anything in my right hand, and right clicked.

Result:  Same as before.  Never was able to "mount" the horse.

(I reread the "horse" article in the Minecraft wiki (
https://minecraft.gamepedia.com/Minecraft_Wiki).  I rediscovered that
golden apples trigger the "love" mode. But that also didn't happen.  I did
feed the horse the 64 golden apples, but It was never tamed.  I grabbed 64
regular apples and fed them to the horse (I was back in Survival), and
again nothing happened.

There's some more information  I don't have any special computer
configurations.  I'm using Windows 10 with the latest updates.  I'm using a
Corsair gaming mouse.  I have used this computer since I started playing
Minecraft in 2015, with the same computer configs, and never had this
problem before.

Via Curse/Twitch, I created a plain Vanilla mod.  I created a new world,
using OneBanana as the name and seed and using a default world.  Finally, I
found some horses, and I was able to mount one--and got bucked off.

I played two other modpacks (MC Eternal and Roguelike Adventures and
Dungeons).  In both, I was able to mount a horse.

I also tried FoolCraft making a default world.  Still can't mount.

Good News.....somewhat.  I downloaded FoolCraft 4, and the anomaly is not
there.  IOW, I can mount a horse.

I am shifting to FC 4.  Looks like you have added some interesting mods.  :)

Hope this helps.

Lawrence

**Comment by Revvilo** [comment]
Yeah, that's really weird. I'm gonna have a look out of sheer interest, so I appreciate how much detail you've gone into. Sounds like something in the pack that I'm just not aware of.
I'm not the developer of the packs, though, hehe, although I'm a fan of the work scalda does. I'm just a player of the pack from a while ago who helps as much as I can on here.
~ All the best :)

**Comment by lawrencekellie** [comment]
Thank you and all the best to you too.

**Comment by Revvilo** [comment]
Hahahah, worked it out. What is happening is that your camera (player) is rotating when you right click the horse (Which *is* vanilla behaviour. Stupid, but vanilla.) BUT then the right-click event is being sent to the server AFTER your camera has rotated. (Which, to be clear, happens the other way around in vanilla) - Which means the server ends up thinking you just right-clicked in the air and thus nothing happens.

You can actually get on the horse if you position yourself so that when your player rotates, you're still aiming at the horse, in which case it plays out as follows: you click, your player rotates but you are still aiming at the horse, the right-click event is sent to the server, and you end up on the horse.

This'll just simply be a mod breaking the event chain (or whatever it is) somewhere along the line.
I hope this was somewhat interesting, although it probably won't get a fix in this pack.

**Comment by lawrencekellie** [comment]
Great!  Fast too.  I'll try it tomorrow morning.

Lawrence
